src/entry.js

```diff
@@ -43,23 +43,27 @@ class Entry {
       clock: new Clock(clockId, clockTime),
     }
 
-    // If signing key was passedd, sign the enrty
-    if (keystore && signKey) {
-      entry = await Entry.signEntry(keystore, entry, signKey) 
-    }
-
+    // If entryValidator was passed, sign the entry (=authorize)
+    entry = await Entry.sign(entry, entryValidator)
     entry.hash = await Entry.toMultihash(ipfs, entry)
     return entry
   }
 
-  static async signEntry (keystore, entry, key) {
-    const signature = await keystore.sign(key, Buffer.from(JSON.stringify(entry)))
+  // TODO: This could instead just return the signature from the entry validator instead of
+  // modifying the entry (modifying parameters isnt a good practice)
```

Comment:
Agreed 👍  We can move the lines that modify the object to line 48 (after signature, before `entry.hash` assignment), then we would do all the modifications in one place.

Reply:
Ok, I can do that on this PR